src/api.js

```diff
@@ -513,81 +517,79 @@ const history: HandlerFunction = async function (req, _res) {
   }
   const verifiedBody = utils.validateHistoryReq(addressesRequestLimit, apiResponseLimit, input);
 
-  switch (verifiedBody.kind) {
-    case "ok": {
-      const body = verifiedBody.value;
-      const limit = apiResponseLimit;
-      const [referenceTx, referenceBlock] = (body.after && [body.after.tx, body.after.block]) || [undefined, undefined];
-      const referenceBestBlock = body.untilBlock;
-
-      const afterBlockNum = await askBlockNum(referenceBlock, referenceTx != undefined ? referenceTx : "");
-      const untilBlockNum = await askBlockNum(referenceBestBlock);
-
-      if (afterBlockNum.kind === 'error') {
-        if (afterBlockNum.errMsg.includes('tx')) {
-          throw new Error("REFERENCE_TX_NOT_FOUND");
-        }
-        if (afterBlockNum.errMsg.includes('block')) {
-          throw new Error("REFERENCE_BLOCK_MISMATCH");
-        }
-        return { status: 400, body: afterBlockNum.errMsg}
-      }
-      if (untilBlockNum.kind === 'error') {
-        throw new Error("REFERENCE_BEST_BLOCK_MISMATCH");
-      }
+  if (verifiedBody.kind === 'error') {
```

Comment:
This is just an unrelated refactor right? I had to copy+paste it into a better diff tool than github to see what was going on.

Reply:
Yeah, there was just a hard-to-read `switch` there, replaced it with a couple of independent small IFs and extracted the main long branch into the function top level. Unrelated to the fix, but was hard to work with it before the change.